src/Controller/UI/DatasetRestrictionsController.php

```diff
@@ -0,0 +1,181 @@
+
+        $datasets = $entityHandler->getBy(Dataset::class, array('id' => $id));
+
+        // RabbitMQ message to update the DOI for the dataset.
+        $rabbitMessage = array(
+            'body' => $id,
+            'routing_key' => 'update'
+        );
+
+        if (!empty($datasets)) {
+            $dataset = $datasets[0];
+            $datasetSubmission = $dataset->getDatasetSubmission();
+            $datasetStatus = $dataset->getDatasetStatus();
+
+            if ($restrictionKey) {
+                // Record the original state for logging purposes before changing it.
+                $from = $datasetSubmission->getRestrictions();
+                $actor = $this->get('security.token_storage')->getToken()->getUser()->getUserId();
+                $this->dispatchLogRestrictionsEvent($dataset, $actor, $from, $restrictionKey);
+
+                $datasetSubmission->setRestrictions($restrictionKey);
+
+                try {
+                    $entityHandler->update($datasetSubmission);
+                } catch (PersistenceException $exception) {
+                    throw new PersistenceException($exception->getMessage());
+                }
+
+                if ($datasetStatus === Dataset::DATASET_STATUS_ACCEPTED) {
+                    $this->publishDoiForAccepted($rabbitMessage);
+                }
+            } else {
+                // Send 500 response code if restriction key is null
+                throw new BadRequestHttpException('Restiction key is null');
+            }
+        }
+        // Send 204(okay) if the restriction key is not null and updated is successful
+
+        return new Response('', 204);
+    }
+
+    /**
+     * Method to publish doi for accepted datasets.
+     *
+     * @param array $rabbitMessage The rabbitMq message that needs to be published.
+     *
+     * @return void
+     */
+    private function publishDoiForAccepted(array $rabbitMessage)
+    {
+       // Publish the message to DoiConsumer to update the DOI.
+
+        //$this->get('old_sound_rabbit_mq.doi_issue_producer')->publish(
+        //    $rabbitMessage['body'],
+        //    $rabbitMessage['routing_key']
+        //);
+    }
+
+    /**
+     * Log restriction changes.
+     *
+     * @param Dataset $dataset          The dataset having restrictions modified.
+     * @param string  $actor            The username of the person modifying the restriction.
+     * @param string  $restrictionsFrom The original restriction.
+     * @param mixed   $restrictionsTo   The restriction that was put in place.
+     *
+     * @return void
+     */
+    protected function dispatchLogRestrictionsEvent(Dataset $dataset, string $actor, string $restrictionsFrom, $restrictionsTo)
+    {
+        $this->logActionItemEventDispatcher->dispatch(
+            array(
+                'actionName' => 'Restriction Change',
+                'subjectEntityName' => $this->entityManager->getClassMetadata(get_class($dataset))->getName(),
```

Comment:
Need to hardcode this to 'Pelagos\Entity\Dataset', because we've previously logged our data with the namespace, changing it would break the reports

Reply:
Yes, we can do it as an improvement after we get done wtih flex 👍